Add API endpoint for featured eligibilities

cliffcrosland opened the pull request:
- Add a new column, `feature_rank`, to the eligibilities table
- Introduce a new API endpoint, `/eligibilities/featured`, that returns:
  - eligibilities to feature on the landing page, sorted by
    `feature_rank`
  - the number of resources associated with each eligibility

Implementation of https://github.com/ShelterTechSF/askdarcel-api/issues/369

lgarofalo commented:
Looks good!

lgarofalo commented:
I'm removing that codefactor configuration, it's not set up well for our ruby for now.

cliffcrosland commented:
@jjfreund Sounds great! Just pushed up the following changes:
- Added service object to get resource_counts for a list of eligibility_ids.
- Added endpoint to update an eligibility.
- Added Postman tests for featured eligibilities and updating eligibility feature_rank.
- Added code to populate eligibilities in local dev env.

Might need some pointers to learn how to properly migrate staging/prod eligibilities to update feature_rank values appropriately to match the spec.

jjfreund commented:
OK, I'll merge this in so the endpoint can be used in front end development -- we can take care of the prod/staging migrations once we get the finalized eligibility names.

derekfidler commented:
Regarding renaming eligibilities that appear on the homepage: MOHCD has requested that we rename 

Ex-Offenders -> Reentry
Young Adults -> Transitional Aged Youth
